### PR TITLE
Create basic Pokedex GUI

### PR DESCRIPTION
**Pull request by pekempy**
**Dashboard.html**
- Modified navbar to add pokedex.html

**data/pokedex.json**
- File generated by data/PokedexGenerator.py
- Lists pokedex for emerald with all encounter data from routes-emerald.json
- Future: Add evolution data to pokedex 

**pokedex.html**
- Dashboard page showing the pokedex, including where 'mons can be obtained on various routes. Pulls data from pokedex.json

**js/pokedex.js**
- Creates the table used in pokedex.html with the relevant data for the 'mon

**routes-emerald.json**
- Added forgotten latios/latias/kyogre/groudon encounters

Pokedex Dashboard Image:
image

### CODE REVIEW

**Comment by pekempy** [comment]
Oh also Nidoran F sprite doesn't work even though there's a replace in there for the file name, worked for M. Not sure why - a maintainer may wanna take a look and fix my implementation of that before merging

**Comment by pekempy** [comment]
UTF-8 encoding fixed but female still causing issues with the replace
image

File: interface/js/pokedex.js
Comment (alexm96) [inline interface/js/pokedex.js:1]: To fix encounter, add something like this here
```
function encounter_log() {
    $.ajax({
            method: "GET",
            url: "http://127.0.0.1:6969/encounter_log",
            crossDomain: true,
            dataType: "json",
            format: "json",
            timeout: 50
        })
        .done(function(encounter_log) {
            reverse_encounter_log = encounter_log["encounter_log"].reverse()
            if (encounter_log["encounter_log"][50]) {
                var range = (moment(reverse_encounter_log[0]["time_encountered"]).subtract(moment(reverse_encounter_log[10]["time_encountered"]))).format("x")
                $('#encounters_hour').text(Math.round((60 / (range / 1000 / 60)) * 10).toLocaleString() + "/h");
            } else {
                $('#encounters_hour').text("-");
            }
        })
}
```

File: interface/js/pokedex.js
Comment (alexm96) [inline interface/js/pokedex.js:153]: And something like this here:
```
window.setInterval(function() {
    encounter_log();
}, 1000);

```

Comment (alexm96) [reply interface/js/pokedex.js:153]: Also probably remove import for dashboard js, alternatively you could probably just add an if check in dashboard.js for the encounter log, which would probably be cleaner. I think it fails when not finding it in the Pokedex, then causing the rest not to run.

**Comment by pekempy** [comment]
dashboard.js now contains fix to check for wrapper. Should be ready to merge now. Also added two Unown sprites (norm/shiny) that were missing

**Comment by 40Cakes** [comment]
Nice, looking good, my fresh bot folder doesn't appear to be loading the page correctly..

image

```
Uncaught TypeError: wrapper is null
    shiny_log file:///C:/Users/.../Documents/GitHub/pekempy/pokebot-bizhawk/interface/js/dashboard.js:351
```

Also, looks like the navbar enocunters/h info will need to be removed from `dashboard.js` and put into it's own file.. As it is now, the pokedex page is still hammering the bot with useless requests and trying to update non-existent elements that only exist in `dashboard.html`

image

**Comment by pekempy** [comment]
Will get that sorted this evening, thanks :)

**Comment by pekempy** [comment]
Updated - Removed dashboard.js script from pokedex.html
Imported the functions from dashboard.js into pokedex.js (May be worth using a header.js and having functions just for the header stats in there given multiple pages are being made?)
Increased the delay for stats_info() and emu_info() to 1000ms from 250ms as it does indeed hammer the api endpoints, reducing to 1s doesn't make a huge difference and the stats are still accurate

**Review by 40Cakes — APPROVED** [review]

**Pull request merged**
